src/components/Inventory/index.tsx

```diff
@@ -146,11 +163,64 @@ export default function Inventory(props: InventoryProps): JSX.Element {
       facilityIds: filters.facilityIds,
       searchSortOrder,
     });
-    const updatedResult = apiSearchResults?.map((result) => {
-      const resultTyped = result as InventoryResult;
-      const facilityInventoriesNames = resultTyped.facilityInventories.map((nursery) => nursery.facility_name);
-      return { ...result, facilityInventories: facilityInventoriesNames.join('\r') };
-    });
+
+    let updatedResult: InventoryResultWithFacilityNames[] | undefined = [];
+    if (filters.facilityIds && filters.facilityIds.length) {
+      const nextResults: InventoryResultWithFacilityNames[] = [];
+      apiSearchResults?.forEach((result) => {
+        const resultTyped = result as FacilityInventoryResult;
+        const indexFound = nextResults.findIndex((res) => res.species_id === resultTyped.species_id);
+
+        if (indexFound !== undefined && indexFound !== -1) {
+          const existingSpecies = nextResults[indexFound];
+          nextResults[indexFound] = {
+            ...existingSpecies,
+            germinatingQuantity: (
+              Number(existingSpecies.germinatingQuantity) + Number(resultTyped['germinatingQuantity(raw)'])
+            ).toString(),
+            notReadyQuantity: (
+              Number(existingSpecies.notReadyQuantity) + Number(resultTyped['notReadyQuantity(raw)'])
+            ).toString(),
+            readyQuantity: (
+              Number(existingSpecies.readyQuantity) + Number(resultTyped['readyQuantity(raw)'])
+            ).toString(),
+            totalQuantity: (
+              Number(existingSpecies.totalQuantity) + Number(resultTyped['totalQuantity(raw)'])
+            ).toString(),
+            facilityInventories: `${existingSpecies.facilityInventories}, ${resultTyped.facility_name}`,
+          };
```

Comment:
In this branch, the `nextResults` aren't added to `updatedResult`. Is that a problem?

Reply:
Yes, we need the results in `updatedResult` to show results in the table. In this branch we add the `nextResults` that have the results when searching with facilityId filters, and in the other (the else) we add the apiSearchResults that have results when searching without facilityId filter applied